### DISCUSSION TITLE
Prefer loading JavaScript in <head>

### DISCUSSION THREAD

**Discussion by miohtama** [pradyunsg/furo#456] [Ideas]
Some complex Sphinx setups rely on JavaScript loading. Most  notable are [nbsphinx](https://nbsphinx.readthedocs.io/en/0.8.9/) and [Plotly](https://plotly.com/). Documentation pages with these options assume you can execute JavaScript in the context body to create interactive widgets.

Because of this, loading JavaScript at the end of the `<body>` does not always work and these add-ons are broken. [Here is an example issue](https://github.com/readthedocs/sphinx_rtd_theme/issues/788#issuecomment-1158705893). To mitigate the issue of difficult JavaScript loading [sphinx-rtd-theme has moved JavaScript back to head](https://github.com/readthedocs/sphinx_rtd_theme/blob/master/sphinx_rtd_theme/layout.html#L62).

While the end of the body JavaScript is recommended by many guides, the current CDNs and JavaScript async loading does not always warrant this - any loading time improvements might be minimal. 

If anyone encounters issues with Sphinx, Furo, nbsphinx and ReadTheDocs, please [prefer to our example documentation how to work around these](https://web3-ethereum-defi.readthedocs.io/).

**Comment by basnijholt** [comment]
I ran into this issue. In the source of your docs I see that the JS loads on top of the HTML.

From your https://github.com/tradingstrategy-ai/web3-ethereum-defi/blame/master/docs/source/conf.py it is not obvious how you solved this. Could you give me some hints?


Edit: I managed to fix it with https://github.com/python-adaptive/adaptive/commit/353b31ee2fea7692b98b687228acf77ba3bcb480.

**Reply by miohtama** [reply]
Sorry for late reply - now say this comment. 

I fixed it by editing the base template directly and fixing the location where JS includes are injected.
